sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -192,7 +192,7 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
     })
   }
 
-  test("unwrap casts when literal is null") {
+  test("SPARK-42741: Do not unwrap casts in binary comparison when literal is null") {
```

Comment:
Expressions in this test are optimized by `NullPropagation`.

Reply:
@wangyum . Although I understand what you mean, this test passed without your patch already.
Could you provide some test case which fails without our patch, @wangyum ?

Reply:
OK. Remove `NullPropagation` in this test.